src/components/sync-connected-sites.tsx

```diff
@@ -249,35 +258,49 @@ const SyncConnectedSitesSection = ( {
 									! pushState.isInProgress &&
 									! pushState.hasFinished && (
 										<div className="flex gap-2 pl-4 ml-auto shrink-0 h-5">
-											<Button
-												variant="link"
-												className="!text-black hover:!text-a8c-blueberry"
-												onClick={ () => {
-													const detail = connectedSite.isStaging
-														? __(
-																"Pulling will replace your Studio site's files and database with a copy from your staging site."
-														  )
-														: __(
-																"Pulling will replace your Studio site's files and database with a copy from your production site."
-														  );
-													showPullConfirmation( () => pullSite( connectedSite, selectedSite ), {
-														detail,
-													} );
-												} }
-												disabled={ isAnySitePulling || isAnySitePushing }
+											<Tooltip
+												text={ getLastSyncTime( selectedSite.id, connectedSite.id, 'pull' ) }
```

Comment:
We are reading the same local storage twice for each connected site in each re-render. One for pull and one for push.

I would suggest using a useState to store those values and a useEffect to store them in the localStorage or appData.

Reply:
I did use it in the updated version. One of my concerns was that when you focus in an out of the app in that case, the time does not update. However, I think the optimization would be worth this.